test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -622,6 +616,44 @@ public async Task ActorOrchestration(bool extendedSessions)
             }
         }
 
+        /// <summary>
+        /// End-to-end test which validates the ContinueAsNew functionality by implementing a counter actor pattern,
+        /// and does so without any waiting between sending events.
+        /// </summary>
+        [Theory]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory + "_BVT")]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task ActorOrchestration_NoWaiting(bool extendedSessions)
+        {
+            using (JobHost host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                nameof(this.ActorOrchestration),
+                extendedSessions))
+            {
+                await host.StartAsync();
+
+                int initialValue = 0;
+                var client = await host.StartOrchestratorAsync(nameof(TestOrchestrations.Counter), initialValue, this.output);
+                await client.RaiseEventAsync("operation", "incr", this.output);
+                await client.RaiseEventAsync("operation", "incr", this.output);
+                await client.RaiseEventAsync("operation", "incr", this.output);
+                await client.RaiseEventAsync("operation", "decr", this.output);
+                await client.RaiseEventAsync("operation", "incr", this.output);
+                await client.RaiseEventAsync("operation", "end", this.output);
+
+                var status = await client.WaitForCompletionAsync(TimeSpan.FromSeconds(10), this.output);
+
+                Assert.Equal(OrchestrationRuntimeStatus.Completed, status?.RuntimeStatus);
+                Assert.Equal(3, (int?)status?.Output);
+
+                // When using ContinueAsNew, the original input is discarded and replaced with the most recent state.
+                Assert.NotEqual(initialValue, status?.Input);

```

Comment:
Are events guaranteed to be processed in the order they are raised? If so, you can strengthen this assertion to make it `Assert.Equal(2, status?.Input)`.

Reply:
That's a very good point. I'll do that.